File: engine/engine-model/src/main/scala/net/server/channel/packet/report/BaseReportPacket.scala
package net.server.channel.packet.report

import net.server.MaplePacket

class BaseReportPacket( private var _type: Int,  private var _victim: String,  private var _reason: Int,  private var _description: String) extends MaplePacket {
     def theType: Int = _type
     def victim: String = _victim
     def reason: Int = _reason
     def description: String = _description
}

File: engine/engine-model/src/main/java/net/server/channel/packet/report/BaseReportPacket.java
package net.server.channel.packet.report;

import net.server.MaplePacket;

public class BaseReportPacket implements MaplePacket {
   private final Integer theType;

   private final String victim;

   private final Integer reason;

   private final String description;

   public BaseReportPacket(Integer theType, String victim, Integer reason, String description) {
      this.theType = theType;
      this.victim = victim;
      this.reason = reason;
      this.description = description;
   }

   public Integer theType() {
      return theType;
   }

   public String victim() {
      return victim;
   }

   public Integer reason() {
      return reason;
   }

   public String description() {
      return description;
   }
}
